### ISSUE TITLE
PySCF fails to compile with GCC 9

### ISSUE THREAD

**Issue by susilehtola** [pyscf/pyscf#329]
PySCF fails to compile with GCC 9 due to broken OpenMP code. For example,
```
/home/work/pyscf/pyscf/lib/np_helper/npdot.c:74:34: error: ‘k’ not specified in enclosing ‘parallel’
   74 |                 int nblk = MAX((k+nthread-1) / nthread, 1);
      |                                 ~^~~~~~~~
/home/work/pyscf/pyscf/lib/np_helper/npdot.c:27:27: note: in definition of macro ‘MAX’
   27 | #define MAX(X,Y)        ((X) > (Y) ? (X) : (Y))
      |                           ^
/home/work/pyscf/pyscf/lib/np_helper/npdot.c:70:9: error: enclosing ‘parallel’
   70 | #pragma omp parallel default(none) shared(a, b, c) \
      |         ^~~
/home/work/pyscf/pyscf/lib/np_helper/npdot.c:76:59: error: ‘m’ not specified in enclosing ‘parallel’
   76 |                 double *cpriv = malloc(sizeof(double) * (m*n+2));
      |                                                          ~^~
/home/work/pyscf/pyscf/lib/np_helper/npdot.c:70:9: error: enclosing ‘parallel’
   70 | #pragma omp parallel default(none) shared(a, b, c) \
      |         ^~~
/home/work/pyscf/pyscf/lib/np_helper/npdot.c:76:59: error: ‘n’ not specified in enclosing ‘parallel’
   76 |                 double *cpriv = malloc(sizeof(double) * (m*n+2));
      |                                                          ~^~
/home/work/pyscf/pyscf/lib/np_helper/npdot.c:70:9: error: enclosing ‘parallel’
   70 | #pragma omp parallel default(none) shared(a, b, c) \
      |         ^~~
```

**Comment by sunqm** [comment]
This error is caused by the changes in OpenMP 4.5 standard.  In the old OpenMP standard (up to v4.0), session 2.14.1.2 states "Variables with const-qualified type having no mutable member are shared.". GCC 9 uses OpenMP 4.5 standard by default and the new standard removes this rule.

Although this error is easy to fix, I'd recommend NOT to use GCC 9 in your production code, for pyscf-1.6, pyscf-1.7. We didn't have enough tests for GCC 9. The changes of OpenMP standard is an example of the problem using GCC 9 series.

**Comment by sunqm** [comment]
Problem is solved by PR #330
